### PR TITLE
Implement native clipping for SvgRenderContext (#1564)

### LINKED ISSUE #1564: Native clipping for SvgExporter
It would be nice if `OxyPlot.SvgRenderContext` could use native SVG clipping using `clip-path`, similar to how the `OxyPlot.SkiaSharp` SVG exporter does it. The functionality seems to be already there in `SvgWriter`, so it seems almost trivial to support this.

Is there any downside why we would not want this?

`SvgRenderContext` is, as far as I know, the only remaining render context without native clipping. If we can support clipping here we could even think about completely removing the complex code path that is there to support renderers without native clipping - but this would be a separate issue.

### PR DESCRIPTION
**Pull request by Jonarw**
Fixes #1564.

### Checklist

- [x] I have included examples or tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file
- [x] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Enable native clipping for SvgRenderContext
- I slightly changed the implementation of clipping in `SvgWriter` (inspired by SkiaSharp SVG output), as the existing implementation did not work well with text
- Added a clipping example to the Rendering Capabilities

@oxyplot/admins

### CODE REVIEW

File: Source/OxyPlot/Svg/SvgRenderContext.cs [hunk 1]
```diff
@@ -262,11 +262,24 @@ public override OxySize MeasureText(string text, string fontFamily, double fontS
             this.w.WriteImage(srcX, srcY, srcWidth, srcHeight, destX, destY, destWidth, destHeight, source);
         }
 
+        /// <inheritdoc/>
+        public override bool SetClip(OxyRect rect)
+        {
+            this.w.BeginClip(rect.Left, rect.Top, rect.Width, rect.Height);
```
Comment (VisualMelon) [inline Source/OxyPlot/Svg/SvgRenderContext.cs:268]: Does this need to call `ResetClip` if we are currently clipping?

File: Source/OxyPlot/Svg/SvgRenderContext.cs [hunk 2]
```diff
@@ -262,11 +262,24 @@ public override OxySize MeasureText(string text, string fontFamily, double fontS
...
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public override void ResetClip()
+        {
+            this.w.EndClip();
```
Comment (VisualMelon) [inline Source/OxyPlot/Svg/SvgRenderContext.cs:275]: Should this (or `EndClip` itself) verify we are currently clipping?

**Comment by Jonarw** [comment]
You are right of course, now I feel a bit silly for not thinking of this...

**Comment by VisualMelon** [comment]
It only occurred to me that it was an issue when I was reading your original SkiaSharp implementation ;)

**Comment by Jonarw** [comment]
I also added a basic test for this, and a `IRenderContext.AutoResetClip` extension because I thought that might come in handy.

**Comment by VisualMelon** [comment]
Looks good to me.

File: Source/Examples/ExampleLibrary/Examples/RenderingCapabilities.cs [hunk 3]
```diff
@@ -17,6 +17,7 @@ namespace ExampleLibrary
     using OxyPlot.Annotations;
     using System.Linq;
     using System.Collections.Generic;
+    using OxyPlot.Axes;
```
Comment (VisualMelon) [inline Source/Examples/ExampleLibrary/Examples/RenderingCapabilities.cs:20]: Is this needed?

Comment (Jonarw) [reply Source/Examples/ExampleLibrary/Examples/RenderingCapabilities.cs]: No, in fact the one above isn't either

File: Source/Examples/ExampleLibrary/Examples/RenderingCapabilities.cs [hunk 4]
```diff
@@ -458,6 +457,59 @@ public static PlotModel DrawTextWithMetrics(string text, string font, double fon
             return model;
         }
 
+        [Example("Clipping")]
+        public static PlotModel Clipping()
+        {
+            var model = new PlotModel();
+            model.Annotations.Add(new DelegateAnnotation(rc =>
+            {
+                // reset a couple of times without any clipping being set to see that nothing explodes
```
Comment (VisualMelon) [inline Source/Examples/ExampleLibrary/Examples/RenderingCapabilities.cs:466]: Ideal

**Pull request merged**